### ISSUE TITLE
tcp_server checking if port is free fails in windows

### ISSUE THREAD

**Issue by zaksnet** [godotengine/godot#35350]
**Godot version:**
3.2 RC1

**OS/device including version:**
Windows 10 

**Issue description:**
When doing a `TCP_Server.listen(somePort)` in windows it fails to check if the port is already in use. Therefore, there is no way to know if a port can be used or not. In the attached project 2 `TCP_Server` instances are starting at the same port (80) and both start without an error.

PS: This has been tested in Linux also and it works as expected. An error is being displayed that the port is already in use.

**Steps to reproduce:**
```GDScript
func _ready():
	var tcp_server = TCP_Server.new()
	var tcp_server2 = TCP_Server.new()
	var err = tcp_server.listen(80)
	yield(get_tree().create_timer(1), "timeout")
	var err2 = tcp_server2.listen(80)
	
	print("Server 1: " + str(err), "\n", "Server 2: " + str(err2))
        # Prints:
        # Server 1: 0
        # Server 2: 0
```
**Minimal reproduction project:**
[DebugProject.zip](https://github.com/godotengine/godot/files/4084768/DebugProject.zip)

**Comment by sumit0190** [comment]
I was under the impression that the socket APIs are mostly shared between Linux and Windows (see `net_socket_posix.cpp` for the implementation); regardless, on the latest Godot I cannot reproduce this. Your project, when run on my Windows 10 system, returns:
```
Server 1: 0
Server 2: 22
```

**Comment by zaksnet** [comment]
> I was under the impression that the socket APIs are mostly shared between Linux and Windows (see `net_socket_posix.cpp` for the implementation); regardless, on the latest Godot I cannot reproduce this. Your project, when run on my Windows 10 system, returns:
> 
> ```
> Server 1: 0
> Server 2: 22
> ```
> 
> This is, of course, different from the Linux results (where both cases return error code `22`), but a similar example coded using pure WinSock also returns the same result.

Results seem to be the same for me with the latest stable version:
![image](https://user-images.githubusercontent.com/1850856/73589433-16415e80-44df-11ea-88e6-1f08b44f2c95.png)
What version of windows are you using? I'm on `Version 1909 (OS Build 18363.592)`

**Comment by sumit0190** [comment]
I'm on version `1809 (OS build 17763.973)`.
![tcp](https://user-images.githubusercontent.com/8823156/73596912-dab18f00-44f4-11ea-978d-2dfeb668e4d5.PNG)

**Comment by zaksnet** [comment]
> I'm on version `1809 (OS build 17763.973)`.
> ![tcp](https://user-images.githubusercontent.com/8823156/73596912-dab18f00-44f4-11ea-978d-2dfeb668e4d5.PNG)

Just tested this on 1809:
![Screenshot_1](https://user-images.githubusercontent.com/1850856/73754866-c0ceb100-476d-11ea-99eb-50efd56171b0.png)
![Screenshot_2](https://user-images.githubusercontent.com/1850856/73754869-c1674780-476d-11ea-8b6c-60808ca5abcc.png)

Port 80 is correctly giving me an error since i'm running an apache server on that PC but port 7899 for example is giving me the same results as Version 1909. So i assume its not a windows issue?

**Comment by zaksnet** [comment]
I just tested this with C# code just to make sure its a not an issue with my PC:
```CSharp
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Net;
using System.Net.Sockets;

namespace WebServerTest
{
    class Program
    {
        static void Main(string[] args)
        {
            try
            {
                TcpListener server = new TcpListener(IPAddress.Any, 1111);
                server.Start();

            }
            catch (Exception ex)
            {
                Console.WriteLine(ex.Message);
            }

            try
            {
                TcpListener server2 = new TcpListener(IPAddress.Any, 1111);
                server2.Start();
            }
            catch (Exception ex)
            {
                Console.WriteLine(ex.Message);
            }
            Console.ReadKey();
        }
    }
}
```

And i get an error as expected:
`Only one usage of each socket address (protocol/network address/port) is normally permitted`

**Comment by Faless** [comment]
@zaksnet can you try out #36321 ?

**Comment by zaksnet** [comment]
Fixed by #36321 🎉🎈
